pkg/resources/helpers.go

```diff
@@ -61,3 +61,41 @@ func setBoolProperty(d *schema.ResourceData, key string, property *sdk.BoolPrope
 	}
 	return nil
 }
+
+func getTagObjectIdentifier(v map[string]any) sdk.ObjectIdentifier {
+	if _, ok := v["database"]; ok {
+		if _, ok := v["schema"]; ok {
+			return sdk.NewSchemaObjectIdentifier(v["database"].(string), v["schema"].(string), v["name"].(string))
+		}
+		return sdk.NewDatabaseObjectIdentifier(v["database"].(string), v["name"].(string))
+	}
+	return sdk.NewAccountObjectIdentifier(v["name"].(string))
```

Comment:
Not sure about this. I think tags are database or database+schema objects and without them it cannot be created / retrieved (unless there's active database or database+schema in the client session), so I should return error here. What do you think ?

Reply:
Hmmm, I think tags are indeed on schema level: https://docs.snowflake.com/en/user-guide/object-tagging#what-is-a-tag and I guess this should be the only option here.

Both schema and database are required in schema resource definition, so it is safe to assume they will be present here.

@scottwinkler, can you also chip in?

Reply:
Hmm, I wonder what would happen if only database would be specified. It would process with "PUBLIC" schema or threw an error.